lib/elixir/lib/kernel.ex

```diff
@@ -3029,16 +3029,23 @@ defmodule Kernel do
     in_module? = (__CALLER__.context == nil)
 
     right = case bootstrapped?(Macro) do
-      true  -> Macro.expand(right, __CALLER__)
+      true  -> Macro.prewalk(right, &Macro.expand(&1, __CALLER__))
```

Comment:
This changes the semantic of the code, for example: `[(alias Foo, as: A), A.bar]`.

I assume you are doing this because of the tail but we should probably expand it only if necessary (i.e. inside in_list when in_module? is false).